tutorials/users/remote-processing/remote-processing.md

```diff
@@ -1,7 +1,199 @@
+needed for that step is hosted. Only the result from the last step in the process will be returned to the user.
+
+Workflows are defined using the [common workflow language (CWL)](https://www.commonwl.org/) which describes which atomic
+processes to execute in what order and how the data should be transferred between steps. Workflows can contain atomic
+processes hosted on different nodes in the network if needed.
+
+## The Weaver python client
+
+The Weaver software comes with a command line interface (CLI) and a python client. In order to run either, you need the
+weaver python package installed. 
+
+This package may already be installed if you're working in a [Marble Jupyterlab](../ide/jupyterlab-howto.md) 
+environment. To test whether the package is installed, run the following command in a JupyterLab console:
+
+```
+import weaver
+```
+
+If the package is not installed you can install it from the git repository with the `pip` command. Run the following
+in a JupyterLab console:
+
+```
+import sys
+import subprocess
+subprocess.run([sys.executable, "-m", "pip", "install", "git+https://github.com/crim-ca/weaver.git"])
+```
+
+### Using the python client
+
+The main component of Weaver is the process and so most of the interaction that you will have with the Weaver API will
+involve inspecting, executing, and managing processes.
+
+To interact with Weaver, first create a `weaver_client` object:
+
+```
+from weaver.cli import WeaverClient
+weaver_client = WeaverClient("https://<your-node-url>/weaver")
+```
+
+```{tip}
+If you are in a [Marble JupyterLab IDE](../ide/ide.html) and you have the 
+[Marble python client](../python-client/python-client.html) installed, you can use this code snippet instead so you
+don't have to explicitly write out the URL.
+
+:::
+from weaver.cli import WeaverClient
+from marble_client import MarbleClient
+marble_client = MarbleClient()
+
+weaver_client = WeaverClient(marble_client.this_node.weaver.url)
+:::
 ```
+
+<!-- TODO: add in authentication info once https://github.com/crim-ca/weaver/pull/599 has been merged -->
+<!-- TODO: add in information about discovering and executing providers as well as processes -->
+
+(weaver-client-discover)=
+#### Discover processes
+
+To discover which processes are available on a Weaver service installed on a Marble node use the
+[`capabilities` method](https://pavics-weaver.readthedocs.io/en/latest/cli.html#getcapabilities-example):
+
+```
+weaver_client.capabilities()
+```
+
+This will return an object containing a list of process names. To inspect the details of each process use the
+[`describe` method](https://pavics-weaver.readthedocs.io/en/latest/cli.html#describeprocess-example)
+
+```
+weaver_client.describe("process_name_here")
+```
+
+The description should include a plain language description of what the process does as well as describing expected
+inputs and outputs. Take note of the inputs especially as that will inform how you can execute this process.
```

Comment:
This sentence is interesting. Would it be possible to elaborate here as to how the properties of the inputs might inform the execution of the processes themselves?

Reply:
I'd love to but I really don't understand myself. Here's the best example I could find of what the output could look like: https://pavics-weaver.readthedocs.io/en/latest/cli.html#describeprocess-example

It looks like a json schema but I really don't understand how to describe how that maps to something like the input example here https://pavics-weaver.readthedocs.io/en/latest/cli.html#id12